R/lint.R

```diff
@@ -268,8 +292,22 @@ lint_dir <- function(path = ".", relative_path = TRUE, ..., exclusions = list("r
 #'   )
 #' }
 #' @export
-lint_package <- function(path = ".", relative_path = TRUE, ...,
-                         exclusions = list("R/RcppExports.R"), parse_settings = TRUE) {
+lint_package <- function(path = ".", ...,
+                         relative_path = TRUE,
+                         exclusions = list("R/RcppExports.R"),
+                         parse_settings = TRUE) {
+  # TODO(next release after 3.0.0): remove this deprecated workaround
+  dots <- list(...)
+  if (length(dots) > 0L && is.logical(dots[[1L]])) {
+    warning(
+      "'relative_path' is no longer available as a positional argument; ",
+      "please supply 'relative_path' as a named argument instead. ",
+      "This warning will be upgraded to an error in the next release."
+    )
+    relative_path <- dots[[1L]]
+    dots <- dots[-1L]
+  }
```

Comment:
Is this tested?

Reply:
no: https://github.com/r-lib/lintr/pull/937#issuecomment-1074703326

Reply:
Let's add one just to be safe.
Rest LGTM.